### PR TITLE
Add CONSTANT_Dynamic to classfile parsing

### PR DESCRIPTION
**Pull request by fengxue-IS**
Adds parsing for Constant Pool type CFR_CONSTANT_Dynamic for
classfile version 55 or higher

Close: #1270

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

**Comment by fengxue-IS** [comment]
Reviewer: @tajila

File: runtime/bcutil/cfreader.c [hunk 1]
```diff
@@ -1128,12 +1128,20 @@ readPool(J9CfrClassFile* classfile, U_8* data, U_8* dataEnd, U_8* segment, U_8*
 			i++;
 			break;
 
+		case CFR_CONSTANT_Dynamic:
+			if (classfile->majorVersion < 55) {
+				errorCode = J9NLS_CFR_ERR_CP_ENTRY_INVALID_BEFORE_V51__ID;
```
Comment (DanHeidinga) [inline runtime/bcutil/cfreader.c]: `J9NLS_CFR_ERR_CP_ENTRY_INVALID_BEFORE_V51__ID`  This should be v55 shouldn't it?

Comment (fengxue-IS) [reply runtime/bcutil/cfreader.c]: @DanHeidinga you are right, I ll add a new message to `cfrerr.nls`, do I need to consider translations?

Comment (DanHeidinga) [reply runtime/bcutil/cfreader.c]: Adding it to the end of the file is sufficient.  It'll get translated in the next translation pass

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk9

**Pull request merged**